### PR TITLE
Tests for running composite under nested aggregation

### PR DESCRIPTION
**Pull request by not-napoleon**
This PR adds REST layer and junit layer tests for running composite aggregation under nested aggregations.  It also adds some REST tests for nested, since I was there and we don't have any right now.

This is related to #38863 but I want to merge these tests separately so I can verify the current behavior before that change.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine update branch

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregatorTests.java [hunk 1]
```diff
@@ -106,13 +116,30 @@ public void setUp() throws Exception {
...
+    }
+
+    @Override
+    protected List<ObjectMapper> objectMappers() {
+        return objectMappers;
+    }
+
+    @Override
+    protected IndexReader wrapDirectoryReader(DirectoryReader reader) throws IOException {
+        if (useNestedDirectoryWrapping) {
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregatorTests.java:139]: Could you check if the list is non-empty?

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregatorTests.java [hunk 2]
```diff
@@ -87,6 +87,21 @@
 import static org.elasticsearch.search.aggregations.AggregationBuilders.nested;
 import static org.hamcrest.Matchers.equalTo;
 
+/**
+ * Tests for the Nested aggregator.
+ *
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregatorTests.java:92]: `<p>`.

Silly javadoc, you are too html.

Comment (not-napoleon) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregatorTests.java:92]: Bad HTML.  `<p>` != `<br/>`.  But fine.

Comment (nik9000) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregatorTests.java:92]: ? I mean, you could `<p>` the whole paragraph. I dunno. whatever works.

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**